### PR TITLE
fix(mail): keep reader navigation and archive actions in sync

### PR DESCRIPTION
**Pull request by elie222**
> ✅ **Browser QA passed** · [QA report](https://qa.getinboxzero.com/20260816-004456_pr-3290_41c3d5d8afc3/report.html)
>
> <sub>Apply `rerun-qa` to rerun.</sub>

## Summary
- Keep keyboard navigation synchronized with the open reader and its position.
- Advance the reader to the next available conversation after archive.
- Cover navigation and removal edge cases with focused unit tests.

## Validation
- `pnpm test app/(app)/[emailAccountId]/mail`
- `pnpm lint`

<a href="https://cubic.dev/pr/elie222/inbox-zero/pull/3290?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/web/app/(app)/[emailAccountId]/mail/thread-list-behavior.ts [hunk 1]
```diff
@@ -2,6 +2,49 @@ export const THREAD_PREFETCH_REMAINING = 8;
...
+  openThreadId: string | null;
+}): number {
+  const clampedFocusedIndex = Math.min(
+    Math.max(0, focusedIndex),
+    Math.max(0, threadIds.length - 1),
+  );
+  if (!openThreadId) return clampedFocusedIndex;
+
+  const openThreadIndex = threadIds.indexOf(openThreadId);
+  return openThreadIndex >= 0 ? openThreadIndex : clampedFocusedIndex;
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/mail/thread-list-behavior.ts]: Fixed in 41c3d5d8a. A missing reader now yields no active row, and action targeting explicitly uses the open thread ID even when a stale selection exists. Covered by focused unit tests.

File: apps/web/app/(app)/[emailAccountId]/mail/MailShell.tsx [hunk 2]
```diff
@@ -315,19 +324,38 @@ export function MailShell() {
...
+      autoAdvanceReader = false,
+    ) => {
       if (isAllAccounts) return;
       const ids = selection.targetIds(
         focusedThread ? getListThreadKey(focusedThread) : undefined,
       );
       if (!ids.length) return;
       if (removeFromList && openThreadId && ids.includes(openThreadId)) {
-        setOpenThreadId(null);
+        setOpenThreadId(
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/app/(app)/[emailAccountId]/mail/MailShell.tsx]: Fixed in 41c3d5d8a. Archive advancement now returns both the surviving thread ID and its post-removal index, and the reader updates both pieces of state. Covered by the earlier-row removal regression test.
